core/src/main/scala/org/apache/spark/deploy/PythonRunner.scala

```diff
@@ -37,8 +38,11 @@ object PythonRunner {
     val pythonFile = args(0)
     val pyFiles = args(1)
     val otherArgs = args.slice(2, args.length)
-    val pythonExec =
-      sys.env.getOrElse("PYSPARK_DRIVER_PYTHON", sys.env.getOrElse("PYSPARK_PYTHON", "python"))
+    val sparkConf = new SparkConf()
+    val pythonExec = sparkConf.get(PYSPARK_DRIVER_PYTHON)
```

Comment:
I find the version I suggested earlier (using `orElse` instead of nested `getOrElse` calls) more readable.